### PR TITLE
Handle errors in playbook requests (eventTypes etc)

### PR DESCRIPTION
**Pull request by RedSquirrelious**
Short circuit in MediatR pipeline.  Try Catch for when playbook down.  Base Controller method that returns NotFound if Try/Catch called & no items returned from Get methods.  If Get methods return items, returns items with 200 status code.

### CODE REVIEW

File: Domain/Sia.Shared.Tests/Extensions/HandlerShortCircuitTests.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Sia.Gateway.Tests.TestDoubles;
+
+namespace Sia.Shared.Tests.Extensions
+{
+    [TestClass]
+    public class HandlerShortCircuitTests
+    {
+        public Task<string> ReturnMockNext()
```
Comment (pdimitratos) [inline Domain/Sia.Shared.Tests/Extensions/HandlerShortCircuitTests.cs]: It should be possible to declare this in the Handle_Should_Return_Next_If_ShouldRequestContinue_Is_True() method as something like

Task<string> ReturnMockNext() => Task.FromResult("Next result");

Which would make it more clear that it's used only in that method.

File: Domain/Sia.Shared.Tests/TestDoubles/MockShortCircuit.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Configuration;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+
+namespace Sia.Gateway.Tests.TestDoubles
+{
+    internal class MockShortCircuit : HandlerShortCircuit<StubRequest, string>
```
Comment (pdimitratos) [inline Domain/Sia.Shared.Tests/TestDoubles/MockShortCircuit.cs]: Now that I think about it, this should be able to be : HandlerShortCircuit<IRequest<string>, string>

Then we could remove the StubRequest class; we always pass null anyway, so it won't change behavior in the tests, just cleans this file up a bit.

File: Domain/Sia.Shared/Extensions/Mediatr/ProxyHandler.cs [hunk 3]
```diff
@@ -57,10 +58,18 @@ public abstract class ProxyHandlerBase<TRequest>
...
-            return await _client.SendAsync(message, cancellationToken);
+            try
+            {
+                var message = new HttpRequestMessage(Method(), RelativeUri(request));
+                AddContentToMessage(request, message);
+                await AddAuthorizationToMessage(request, message);
+                var result = await _client.SendAsync(message, cancellationToken);
+                return result;
+            }
+            catch (HttpRequestException ex)
```
Comment (pdimitratos) [inline Domain/Sia.Shared/Extensions/Mediatr/ProxyHandler.cs]: Does this exception get logged anywhere? If not yet, we should try to do so.

File: Domain/Sia.Shared/Extensions/Mediatr/MockEventTypeData.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Sia.Shared.Extensions.Mediatr
+{
+    public class MockEventTypeData
```
Comment (pdimitratos) [inline Domain/Sia.Shared/Extensions/Mediatr/MockEventTypeData.cs]: Did this end up getting used anywhere?

File: Domain/Sia.Shared.Tests/TestDoubles/MockShortCircuit.cs (hunk 2 above)
Comment (RedSquirrelious) [reply Domain/Sia.Shared.Tests/TestDoubles/MockShortCircuit.cs]: Changing it to IRequest gets you this error:  "The type MediatR.IRequest must be convertible to MediatR.IRequest<string> in order to use it as a parameter ..."

File: Domain/Sia.Shared/Extensions/Mediatr/MockEventTypeData.cs (hunk 4 above)
Comment (RedSquirrelious) [reply Domain/Sia.Shared/Extensions/Mediatr/MockEventTypeData.cs]: yes, in the Gateway.Requests.Playbook.EventTypes.GetEventType GetEventTypeShortCircuit 
method GenerateMockAsync

File: Domain/Sia.Shared/Extensions/Mediatr/ProxyHandler.cs (hunk 3 above)
Comment (RedSquirrelious) [reply Domain/Sia.Shared/Extensions/Mediatr/ProxyHandler.cs]: we talked about putting logging in a subsequent PR.  want it to come in earlier?

**Pull request merged**